jetstream/experimenter.py

```diff
@@ -29,32 +81,101 @@ class Experiment:
     variants: List[Variant]
     normandy_slug: Optional[str] = None
 
-    @staticmethod
-    def _unix_millis_to_datetime(num: Optional[float]) -> Optional[dt.datetime]:
-        if num is None:
-            return None
-        return dt.datetime.fromtimestamp(num / 1e3, pytz.utc)
+    @classmethod
+    def from_dict(cls, d) -> "LegacyExperiment":
+        converter = cattr.Converter()
+        converter.register_structure_hook(
+            dt.datetime, lambda num, _: _unix_millis_to_datetime(num),
+        )
+        return converter.structure(d, cls)
+
+    def to_experiment(self) -> "Experiment":
+        """Convert to Experiment."""
+        branches = [Branch(slug=variant.slug, ratio=variant.ratio) for variant in self.variants]
+        control_slug = None
+
+        control_slugs = [variant.slug for variant in self.variants if variant.is_control]
+        if len(control_slugs) == 1:
+            control_slug = control_slugs[0]
+
+        return Experiment(
+            normandy_slug=self.normandy_slug,
+            slug=self.slug,
+            type=self.type,
+            status=self.status,
+            active=self.status == "Live",
+            start_date=self.start_date,
+            end_date=self.end_date,
+            proposed_enrollment=self.proposed_enrollment,
+            features=[],
+            branches=branches,
+            reference_branch=control_slug,
+        )
+
+
+@attr.s(auto_attribs=True, kw_only=True, slots=True, frozen=True)
+class NimbusExperiment:
+    """Represents a v4 experiment from Experimenter."""
+
+    slug: str  # Normandy slug
+    active: bool
+    features: List[Feature]
+    branches: List[Branch]
+    start_date: dt.datetime
+    end_date: Optional[dt.datetime]
+    proposed_enrollment: int
+    reference_branch: str
 
     @classmethod
-    def from_dict(cls, d) -> "Experiment":
+    def from_dict(cls, d) -> "NimbusExperiment":
         converter = cattr.Converter()
         converter.register_structure_hook(
-            dt.datetime, lambda num, _: cls._unix_millis_to_datetime(num),
+            dt.datetime, lambda num, _: _unix_millis_to_datetime(num),
         )
         return converter.structure(d, cls)
 
+    def to_experiment(self) -> "Experiment":
+        """Convert to Experiment."""
+        return Experiment(
+            normandy_slug=None,
+            slug=self.slug,
+            type="nimbus",
```

Comment:
I'm not sure about this. Nimbus experiments don't seem to have a type specified, I'm assigning a type here since we are filtering experiments based on type when we determine what experiments to analyse.

Reply:
Beware that these experiments are included as "rapid"-type experiments in the v1 API. Let's use type="v4" to disambiguate here.